File: 882-peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int n=arr.size();
             int start=0;
             int end=n-1;
             while(start<end){
                // int mid=start+(end-start)/2;
                int mid=(start+end)/2;
                if(arr[mid]<arr[mid+1]){
                    start=mid+1;
                }
                else{
                    end=mid;
                }
                // mid=start+(end-start)/2;
                  mid=(start+end)/2;
             }
             return start;
    }

};

File: 882-peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
         int max = Integer.MIN_VALUE, idx=0;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>max){
                max=arr[i];
                idx=i;
            }
        }
        return idx;
    }
}